webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx

```diff
@@ -0,0 +1,287 @@
+        column.render = checkpointRenderer;
+      } else if (column.key === V1GetExperimentCheckpointsRequestSortBy.STATE) {
+        column.filterDropdown = stateFilterDropdown;
+        column.isFiltered = (settings) => !!(settings as Settings).state;
+        column.filters = ([ 'ACTIVE', 'COMPLETED', 'DELETED', 'ERROR' ] as CheckpointState[])
+          .map((value) => ({
+            text: <Badge state={value} type={BadgeType.State} />,
+            value,
+          }));
+      } else if (column.key === 'actions') {
+        column.render = actionRenderer;
+      }
+      if (column.key === settings.sortKey) {
+        column.sortOrder = settings.sortDesc ? 'descend' : 'ascend';
+      }
+      return column;
+    });
+
+    return newColumns;
+  }, [ dropDownOnTrigger, experiment, settings.sortDesc, settings.sortKey, stateFilterDropdown ]);
+
+  const handleTableChange = useCallback((tablePagination, tableFilters, tableSorter) => {
+    if (Array.isArray(tableSorter)) return;
+
+    const { columnKey, order } = tableSorter as SorterResult<CoreApiGenericCheckpoint>;
+    if (!columnKey || !columns.find((column) => column.key === columnKey)) return;
+
+    const newSettings = {
+      sortDesc: order === 'descend',
+      /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+      sortKey: columnKey as any,
+      tableLimit: tablePagination.pageSize,
+      tableOffset: (tablePagination.current - 1) * tablePagination.pageSize,
+    };
+    const shouldPush = settings.tableOffset !== newSettings.tableOffset;
+    updateSettings(newSettings, shouldPush);
+  }, [ columns, settings.tableOffset, updateSettings ]);
+
+  const fetchExperimentCheckpoints = useCallback(async () => {
+    try {
+      const states = (settings.state ?? []).map((state) => (
+        encodeCheckpointState(state as CheckpointState)
+      ));
+      const response = await getExperimentCheckpoints(
+        {
+          id: experiment.id,
+          limit: settings.tableLimit,
+          offset: settings.tableOffset,
+          orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
+          sortBy: validateDetApiEnum(V1GetExperimentCheckpointsRequestSortBy, settings.sortKey),
+          states: validateDetApiEnumList(Determinedcheckpointv1State, states),
+        },
+        { signal: canceler.signal },
+      );
+      setTotal(response.pagination.total ?? 0);
+      setCheckpoints(response.checkpoints);
+    } catch (e) {
+      handleError(e, {
+        publicSubject: `Unable to fetch experiment ${experiment.id} checkpoints.`,
+        silent: true,
+        type: ErrorType.Api,
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  }, [
+    experiment.id,
+    canceler,
+    settings.sortDesc,
+    settings.sortKey,
+    settings.state,
+    settings.tableLimit,
+    settings.tableOffset,
+  ]);
+
+  const submitBatchAction = useCallback(async (action: CheckpointAction) => {
+    try {
+      // TODO: Actions
```

Comment:
turn off row selection/ `TableBatch` for now?

<img width="214" alt="image" src="https://user-images.githubusercontent.com/28383080/186880758-06f98722-212b-4b7f-98b5-f4cad0d358a3.png">


batch registering of checkpoints would be pretty cool though IMO.

we could generate names for the model versions based on a user input + the different UUIDs or similar.

Reply:
I disabled it for now, I can go back in a later PR and turn this on.